tools/recetox_aplcms/recetox_aplcms_two_step_hybrid.xml

```diff
@@ -0,0 +1,111 @@
+
+            two_step_hybrid_main(
+                sample_files = c('$filenames_str'),
+                known_table_file = '${known_table}',
+                updated_known_table_file = '${updated_known_table}',
+                recovered_file = '${recovered_feature_sample_table}',
+                aligned_file = '${aligned_feature_sample_table}',
+                out_format = '$output_format.out_format',
+                metadata = '${metadata}',
+                cluster = as.integer(Sys.getenv('GALAXY_SLOTS', unset = 1)),
+                min.within.batch.prop.detect = $multibatch_processing.min_within_batch_prop_detect,
+                min.within.batch.prop.report = $multibatch_processing.min_within_batch_prop_report,
+                min.batch.prop = $multibatch_processing.min_batch_prop,
+                batch.align.mz.tol = $multibatch_processing.batch_align_mz_tol,
+                batch.align.chr.tol = $multibatch_processing.batch_align_chr_tol,
+                min.pres = $noise_filtering.min_pres,
+                min.run = $noise_filtering.min_run,
+                mz.tol = $noise_filtering.mz_tol,
+                baseline.correct.noise.percentile = $noise_filtering.baseline_correct_noise_percentile,
+                baseline.correct = $noise_filtering.baseline_correct,
+                intensity.weighted = $noise_filtering.intensity_weighted,
+                shape.model = '$feature_detection.shape_model',
+                peak.estim.method = '$feature_detection.peak_estim_method',
+                BIC.factor = $feature_detection.BIC_factor,
+                min.bw = $feature_detection.min_bandwidth,
+                max.bw = $feature_detection.max_bandwidth,
+                sd.cut = c($feature_detection.sd_cut_min, $feature_detection.sd_cut_max),
+                sigma.ratio.lim = c($feature_detection.sigma_ratio_lim_min, $feature_detection.sigma_ratio_lim_max),
+                component.eliminate = $feature_detection.component_eliminate,
+                moment.power = $feature_detection.moment_power,
+                align.mz.tol = $peak_alignment.align_mz_tol,
+                align.chr.tol = $peak_alignment.align_chr_tol,
+                max.align.mz.diff = $peak_alignment.max_align_mz_diff,
+                recover.mz.range = $weak_signal_recovery.recover_mz_range,
+                recover.chr.range = $weak_signal_recovery.recover_chr_range,
+                use.observed.range = $weak_signal_recovery.use_observed_range,
+                recover.min.count = $weak_signal_recovery.recover_min_count,
+                match.tol.ppm = $history_db.match_tol_ppm,
+                new.feature.min.count = $history_db.new_feature_min_count
+            )
+        ]]></configfile>
+    </configfiles>
+
+    <expand macro="inputs">
+        <param name="metadata" type="data" format="csv" label="metadata" help="A csv table with two columns, where the first column contains sample names and the second batch labels for each sample."/>
```

Comment:
Is this table aligned with what WaveICA and RAMClustR expect as metadata table? I think it would be beneficial to just have a single table type for metadata.

Reply:
Yes, the tool will work with the metadata table described here: https://github.com/RECETOX/galaxytools/issues/132